src/games/strategy/triplea/ui/AbstractUIContext.java

```diff
@@ -297,33 +297,24 @@ public void shutDown() {
     return rVal;
   }
 
-  private static Map<String,String> getSkins(final String mapName) {
+  private static Map<String, String> getSkins(final String mapName) {
     final Map<String, String> rVal = new HashMap<>();
     final File[] files = ClientFileSystemHelper.getUserMapsFolder().listFiles();
     if (files == null) {
       return rVal;
     }
-
     for (final File f : files) {
-      if (!f.isDirectory()) {
-        if (mapSkinNameMatchesMapName(f.getName(), mapName)) {
-          final String skinName = f.getName().substring(0, f.getName().indexOf('-'));
-          if (f.getName().endsWith(".zip")) {
-            rVal.put(skinName, f.getName());
-          } else {
-            rVal.put(f.getName().substring(f.getName().indexOf('-') + 1), f.getName());
-          }
-        }
+      if (mapSkinNameMatchesMapName(f.getName(), mapName)) {
+        final String displayName = f.getName().replace(mapName + "-", "").replace("-master", "").replace(".zip", "");
```

Comment:
This does not look correct, if "-" is removed first, then "-master" will never match anything.

Some commentary around this code would be useful. This loading logic has gotten complicated enough, we would really benefit from some tests. Its' been very difficult to update something and not cause a regression.